src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -875,6 +875,18 @@ private VariableDesc(String name) {
             this(name, null, null);
         }
 
+        /**
+         * Create a new VariableDesc instance.
+         *
+         * @param name name of the variable
+         * @param scope ast of type {@link TokenTypes#SLIST} or
+         *              {@link TokenTypes#LITERAL_FOR} or {@link TokenTypes#OBJBLOCK}
+         *              which is enclosing the variable
+         */
+        private VariableDesc(String name, DetailAST scope) {
+            this(name, null, scope);
+        }
+
```

Comment:
Ok for killing the mutation. IMO the check design should be slightly modified not to ask for `type` if it is an instance variable.